simulation.cpp

```diff
@@ -389,21 +390,29 @@ void test_simulation() noexcept//!OCLINT test may be many
 
         simulation s{all_params{
                 identity_env,
-                identity_ind,
+                potental_identity_ind,
                 minimal_pop,
                 sim_p
             }};
 
+        //give simulation a simple input
+        //this will be used to calculate optimum
+        //and also fed to individuals
+        s.update_inputs({1});
+        //give inputs to inds
+        assign_inputs(s);
+
         //change target value to match output of ind 0 net
         size_t best_ind = 0;
-        change_nth_ind_net(s, best_ind, potential_identity_net);
+        change_nth_ind_net(s, best_ind, identity_net);
         auto best_net = get_nth_ind_net(s, best_ind);
 
         size_t worst_ind = 1;
+        change_nth_ind_net(s, worst_ind, change_all_weights(potential_identity_net,-1));
         auto worst_net = get_nth_ind_net(s,worst_ind);
 
         assert(net_behaves_like_the_function(best_net, identity_env.env_function_A));
-
+        assert(!net_behaves_like_the_function(worst_net, identity_env.env_function_A));
 
         select_inds(s);
 
```

Comment:
I must say I am quite confused at all these changes, they seem to me to not really change the function of the test, and to only make it a bit more confusing to read? I mean I'm not against them, just not quite getting the point.

Reply:
Apart from the last assert, which is good.

Reply:
it was me debugging you can remove them if you like, although I think they actually make the test clearer, because worst net should not behave like the function

Reply:
ah ok you refer to the other changes. Do you mean the renaming of things like potentially_identity etc? I would keep them.

Reply:
Yeah we can keep them. I'm just a bit confused, you said you had rewritten this test but I don't see which part of what has changed actually changes anything about how it works?